src/Avalonia.Base/DirectProperty.cs

```diff
@@ -156,7 +157,10 @@ public class DirectProperty<TOwner, TValue> : DirectPropertyBase<TValue>, IDirec
         /// <inheritdoc/>
         internal override TValue InvokeGetter(IAvaloniaObject instance)
         {
-            return Getter((TOwner)instance);
+            var value = Getter((TOwner)instance);
+            if (EqualityComparer<TValue>.Default.Equals(value, default))
+                return GetUnsetValue(instance.GetType());
```

Comment:
This would make calling `control.GetValue(MyDirectValueProperty)` return a different value to `control.MyDirectValue`.